### PR TITLE
Add a functionality for getting a photo from a contact

### PR DESCRIPTION
**Pull request by vinothvino42**
**EDL:**

```yaml
View:
  header:
    title: PhoneContacts
  styles:
    scrollableView: false
  body:
    Column:
      styles:
        gap: 24
        padding: 24
        mainAxis: center
        crossAxis: center
      children:
        - Button:
            label: Get Phone Contacts
            onTap:
              getPhoneContacts:
                onSuccess: |
                  //@code
                  console.log("Phone Number Fetched Successfully");
                  ensemble.storage.set('contacts', event.data['contacts']);

                onError: |
                  //@code
                  console.log("Something went wrong!");
        - ListView:
            id: listView
            styles: { expanded: true }
            item-template:
              data: ${ensemble.storage.contacts}
              name: contact
              template:
                UserContact:
                  inputs:
                    contact: ${contact}

UserContact:
  inputs:
    - contact
  styles: { gap: 10 }
  onLoad:
    getPhoneContactPhoto:
      id: contactPhotoId
      contactId: ${contact.id}
  body:
    Column:
      styles: { gap: 2 }
      children:
        - Row:
            children:
              - Image:
                  source: ${contactPhotoId.image}
                  width: 100
                  height: 100
                  fallback: SmallFallbackWidget
              - Column:
                  styles: { gap: 2, margin: 10 }
                  children:
                    - Text:
                        textStyle: { fontWeight: bold, fontSize: 18 }
                        text: "ID: ${contact.id}"
                    - Text:
                        textStyle: { fontWeight: bold, fontSize: 18 }
                        text: "Name: ${contact.displayName}"
                    - Text:
                        textStyle: { fontWeight: bold, fontSize: 18 }
                        text: "Phone: ${getPhoneNumber(contact.phones)}"
                    - Shape:
                        styles:
                          height: 10
        - Divider:

SmallFallbackWidget:
  Image:
    styles: { fit: cover, height: 200, width: 200 }
    source: https://phoenixcellular.com/wp-content/uploads/2022/09/overview__bcphzsdb4fpu_og.png

Global: |
  //@code

  if (ensemble.storage.contacts == null) {
    ensemble.storage.contacts = [];
  }

  function getPhoneNumber(phones) {
    if (phones.length > 0) {
      return phones[0].number;
    } else {
      return "Not Found";
    }
  }

```

### CODE REVIEW

**Review by vusters — APPROVED** [review]

**Comment by kmahmood74** [comment]
@vinothvino42 what is this for? Please merge and close if done

**Comment by vinothvino42** [comment]
> @vinothvino42 what is this for? Please merge and close if done

It's not fully completed, Just added the stub here. Need to work on the ensemble_contacts package after that I'll merge this one

**Comment by vinothvino42** [comment]
@vusters I updated it. Can you review it one more time and see the above EDL?

**Review by vusters — APPROVED** [review]
minor changes requested. You can fix and merge in.

File: lib/action/phone_contact_action.dart [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  final EnsembleAction? onSuccess;
+  final EnsembleAction? onError;
+
+  EnsembleAction? getOnSuccess(DataContext dataContext) =>
+      dataContext.eval(onSuccess);
+
+  EnsembleAction? getOnError(DataContext dataContext) =>
+      dataContext.eval(onError);
+
+  factory PhoneContactAction.fromYaml({Invokable? initiator, Map? payload}) {
```
[the remark this answers is not in the document]
Comment (vinothvino42) [reply lib/action/phone_contact_action.dart:27]: Done

File: lib/action/phone_contact_action.dart [hunk 2]
```diff
@@ -0,0 +1,110 @@
+import 'dart:typed_data';
+
+import 'package:ensemble/framework/action.dart';
+import 'package:ensemble/framework/data_context.dart';
+import 'package:ensemble/framework/error_handling.dart';
+import 'package:ensemble/util/utils.dart';
+import 'package:ensemble_ts_interpreter/invokables/invokable.dart';
+
+class PhoneContactAction extends EnsembleAction {
```
[the remark this answers is not in the document]
Comment (vinothvino42) [reply lib/action/phone_contact_action.dart:9]: Done

File: lib/screen_controller.dart [hunk 3]
```diff
@@ -601,6 +614,15 @@ class ScreenController {
     }
   }
 
+  void updateContactData(PhoneContactPhotoAction action,
```
[the remark this answers is not in the document]
Comment (vinothvino42) [reply lib/screen_controller.dart:622]: Done

File: lib/screen_controller.dart [hunk 4]
```diff
@@ -332,10 +333,22 @@ class ScreenController {
             context,
             scopeManager,
             action.getOnError(scopeManager.dataContext)!,
-            event: EnsembleEvent(action.initiator!, data: {'error': error}),
+            event: EnsembleEvent(action.initiator!, error: error),
           );
         }
       });
+    } else if (action is PhoneContactPhotoAction) {
```
[the remark this answers is not in the document]
Comment (vinothvino42) [reply lib/screen_controller.dart:340]: Done

**Comment by kmahmood74** [comment]
what's the ticket number?

**Pull request merged**